### PR TITLE
Reddit confessions dataset

### PR DESCRIPTION
**Pull request by shahules786**
As mentioned in #2044 Augmented dataset for training safety models. 
- The dataset is made in the same format as prosocial-dialog for ease of use. 
- safety label confidence is added to enable users to filter data according to a threshold of one's own preference.

### CODE REVIEW

**Comment by shahules786** [comment]
@bitplane @Vechtomov Can you guys take a look at this PR? I am already using this data to train safety models.

**Comment by shahules786** [comment]
@Vechtomov I see, that makes sense. I can add a `safety datasets` folder and document the format and datasets that are currently used for the purpose. What do you think?

**Comment by Vechtomov** [comment]
Yes, please. I don't think we need separate safety datasets folder, for now changes in `__init.py__` and readme are enough.

**Comment by shahules786** [comment]
@Vechtomov Done.

File: data/datasets/README.md [hunk 1]
```diff
@@ -48,6 +48,18 @@ proposed. The text-only dataset must include the following columns:
 2. **SOURCE** (string)
 3. **METADATA** (JSON string, optional)
 
+### **Prosocial format**
```
Comment (Vechtomov) [inline data/datasets/README.md:51]: I prefer "Safety format" if it is common for all future safety datasets

File: data/datasets/README.md [hunk 2]
```diff
@@ -48,6 +48,18 @@ proposed. The text-only dataset must include the following columns:
...
+
+For datasets that are intended to be used to train safety models, prosocial
+format is proposed. The format is given below
+
+1. **USER** (string)
+2. **RESPONSE** (string, optional)
+3. **ROTs** (List)
+4. **SAFETY_LABEL** (string - categorical)
+5. **EPISODE_DONE** (bool)
+6. **SOURCE** (string,optional)
```
Comment (Vechtomov) [inline data/datasets/README.md:61]: This needs more detailed description. I didn't realize it until I checked ProsocialDialog paper.

File: data/datasets/__init__.py [hunk 3]
```diff
@@ -13,3 +13,8 @@
     "youtube_subs_howto100M": "totuta/youtube_subs_howto100M",
     "iapp_wiki_qa_squad": "wannaphong/iapp_wiki_qa_squad_oa",
 }
+
+SAFETY_DATASETS = {
+    "prosocial-dialog": "allenai/prosocial-dialog",
+    "prosocial-confessions": "shahules786/prosocial-confessions",
```
Comment (Vechtomov) [inline data/datasets/__init__.py:19]: How these two datasets can be fit in one training process if `RESPONSE` column isn't provided in `prosocial-confessions` dataset?

**Review by Vechtomov — CHANGES_REQUESTED** [review]

File: data/datasets/__init__.py (hunk 3 above)
Comment (shahules786) [reply data/datasets/__init__.py:19]: The current response for the user's prompt is not taken into consideration while modeling. Response is used to form contextual information about past conversations. In prosocial confessions, each example is a single conversation indicated by `episode_done` set to True.

File: data/datasets/README.md (hunk 2 above)
Comment (shahules786) [reply data/datasets/README.md]: @Vechtomov  Done.

Comment (Vechtomov) [reply data/datasets/README.md]: Can you also add a link to the paper?

**Review by Vechtomov — APPROVED** [review]

**Pull request merged**